Content.Server/Sports/Components/BaseballBatComponent.cs

```diff
@@ -0,0 +1,56 @@
+using Robust.Shared.Audio;
+
+namespace Content.Server.Sports.Components
+{
+
+    /// <summary>
+    /// Give this to a melee weapon with wide attack and it will be able to bat thrown objects
+    /// </summary>
+    [RegisterComponent]
+    public sealed class BaseballBatComponent : Component
+    {
+        /// <summary>
+        /// The minimum amount of extra distance from that the item being hit will travel.
+        /// </summary>
+        [DataField("wackForceMultiplierMin")]
+        [ViewVariables(VVAccess.ReadWrite)]
+        public float WackForceMultiplierMin {get; set; } = 1f;
+
+        /// <summary>
+        /// The maximum amount of extra distance from that the item being hit will travel.
+        /// </summary>
+        [DataField("wackForceMultiplierMax")]
+        [ViewVariables(VVAccess.ReadWrite)]
+        public float WackForceMultiplierMax {get; set; } = 8f;
+
+        /// <summary>
+        /// The minimum amount of velocity that the item being hit should be given
+        /// </summary>
+        [DataField("wackStrengthMin")]
+        [ViewVariables(VVAccess.ReadWrite)]
+        public float WackStrengthMin {get; set; } = 10f;
+
+        /// <summary>
+        /// The maximum amount of velocity that the item being hit should be given
+        /// </summary>
+        [DataField("wackStrengthMax")]
+        [ViewVariables(VVAccess.ReadWrite)]
+        public float WackStrengthMax {get; set; } = 15f;
+
+        /// <summary>
+        /// Sound that plays when you have a good hit.
+        /// A good hit is when the item is going to go at least as far as the target
+        /// </summary>
+        [DataField("hitSound")]
+        [ViewVariables(VVAccess.ReadOnly)]
+        public SoundSpecifier HitSound = new SoundPathSpecifier("/Audio/Effects/baseball-hit.ogg");
+
+        /// <summary>
+        /// Chances of the item being hit turning into a fireball.
+        /// Should be a very high number. Set to 1 if you always want a fireball
+        /// </summary>
+        [DataField("fireballChance")]
+        [ViewVariables(VVAccess.ReadWrite)]
+        public int FireballChance {get; set; } = 5000;
```

Comment:
Chances are floats in the codebase and the lower they are the less probable something is, with "1" being 100% chance. This is way more intuitive so please change it to a float and tweak the check logic in the system accordingly (you can use `Robust.Shared.Random.RandomExtensions.Prob()` for this purpose)